Scripts/WikiUpdater.cs

```diff
@@ -0,0 +1,415 @@
+                }
+            }
+        }
+
+        var reader = File.ReadLinesAsync(ENGLISH_TRANSLATION_FILE, Encoding.UTF8, cancellationToken);
+        var writer = new StreamWriter(File.Create(TEMP_TRANSLATION_FILE), new UTF8Encoding(false));
+
+        var isReplacingValue = false;
+        var isFuzzyValue = false;
+        await foreach (var line in reader)
+        {
+            if (line == string.Empty)
+            {
+                // Blank lines mark the end of a value we might be replacing, so stop replacing
+                isReplacingValue = false;
+                await writer.WriteLineAsync(line);
+                continue;
+            }
+
+            if (isReplacingValue)
+            {
+                // Skip lines for values we're replacing
+                continue;
+            }
+
+            if (line == "#, fuzzy")
+            {
+                // Defer adding fuzzy labels so we can include/exclude them based on the next line
+                isFuzzyValue = true;
+                continue;
+            }
+
+            if (!line.Contains("msgid \""))
+            {
+                // Copy existing lines that we don't want to replace
+                await writer.WriteLineAsync(line);
+                continue;
+            }
+
+            var key = line.Split("\"")[1];
+            if (translationPairs.TryGetValue(key, out var value))
+            {
+                // Skip fuzzy labels if present, since we know this value
+                isFuzzyValue = false;
+                isReplacingValue = true;
+
+                // Add the key line
+                await writer.WriteLineAsync(line);
+
+                // Add the value line(s)
+                var linesToInsert = value.Split("\n");
+                if (linesToInsert.Length == 1)
+                {
+                    await writer.WriteLineAsync($"msgstr \"{value}\"");
+                }
+                else
+                {
+                    await writer.WriteLineAsync("msgstr \"\"");
+
+                    // Split the content over multiple lines with correct formatting
+                    for (var j = 0; j < linesToInsert.Length; j++)
+                    {
+                        var lineToInsert = linesToInsert[j];
+                        var isLastLine = j == linesToInsert.Length - 1;
+                        await writer.WriteLineAsync(isLastLine ? $"\"{lineToInsert}\"" : $"\"{lineToInsert}\\n\"");
+                    }
+                }
+            }
+            else
+            {
+                if (isFuzzyValue)
+                {
+                    // Re-insert the fuzzy label, since it's not a value we know from the wiki
+                    await writer.WriteLineAsync("#, fuzzy");
+                    isFuzzyValue = false;
+                }
+
+                await writer.WriteLineAsync(line);
+            }
+        }
+
+        writer.Dispose();
+        File.Copy(TEMP_TRANSLATION_FILE, ENGLISH_TRANSLATION_FILE, true);
```

Comment:
Couldn't this use move with overwrite target file? That way a copy of the file is not needed to be made.